sdk/core/azure-core/src/environment_log_level_listener.cpp

```diff
@@ -120,15 +120,16 @@ EnvironmentLogLevelListener::GetLogListener()
 
   static std::function<void(Logger::Level level, std::string const& message)> const consoleLogger =
       [](auto level, auto message) {
-        std::ostream* os = &std::cout;
-        if (level == Logger::Level::Error)
+        if (message.empty())
```

Comment:
I think `Log::Write()` should be doing this, so we can say if log message is empty, no log level listener is ever going to be invoked.

Reply:
That's a really good question. And yeah, it makes sense to move this to Log::Write instead of the actual low level logger - at the same point it filters on log level.